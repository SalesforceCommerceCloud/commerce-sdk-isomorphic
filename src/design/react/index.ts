/*
 * Copyright (c) 2025, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

// Main Provider
export {
  PageDesignerProvider,
  usePageDesignerMode,
} from './context/PageDesignerProvider';

export {useDesignContext} from './context/DesignContext';
export type {DesignContextType} from './context/DesignContext';

// Components
export {createReactComponentDesignDecorator} from './components/ComponentDecorator';
export {createReactRegionDesignDecorator} from './components/RegionDecorator';
