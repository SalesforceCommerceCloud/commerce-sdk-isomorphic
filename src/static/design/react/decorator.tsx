/*
 * Copyright (c) 2025, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */
import React from 'react';

export function createReactDesignDecorator(): (
  component: React.ReactElement
) => React.ReactElement {
  // TODO: Implement design decorator for React components.
  return (component: React.ReactElement) => <>{component}</>;
}
