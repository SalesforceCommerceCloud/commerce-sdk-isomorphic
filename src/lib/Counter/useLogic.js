import {useState} from 'react';

const useLogic = initialState => {
  const [count, setCount] = useState(initialState);
  return {
    count,
    incrementCount: () => setCount(count + 1),
  };
};

export default useLogic;
