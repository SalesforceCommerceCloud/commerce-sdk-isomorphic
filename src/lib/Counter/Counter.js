import React from 'react';
import PropTypes from 'prop-types';
import useLogic from './useLogic';
import classes from './Counter.module.css';

const Counter = ({initialValue}) => {
  const {count, incrementCount} = useLogic(initialValue);

  return (
    <div className={classes.counter}>
      <h2 className={classes.header}>Counter</h2>
      <button className={classes.button} type="button" onClick={incrementCount}>
        Increment by one
      </button>
      <div>
        Total value: <strong>{count}</strong>
      </div>
    </div>
  );
};

Counter.propTypes = {
  /**
   * Define initial value for the Counter
   */
  initialValue: PropTypes.number.isRequired,
};

export default Counter;
