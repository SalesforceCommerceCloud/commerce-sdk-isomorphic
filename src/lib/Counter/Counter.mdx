---
name: 'Example Component'
route: '/example'
---
import { Playground, Props } from 'docz'
import Counter from './Counter';

# Example Component Demo

## Counter props
<Props of={Counter} />

## Description

`Counter` component provides user basic interface to increment counter by one.

## Demo

<Playground>
  <Counter initialValue={0} />
</Playground>