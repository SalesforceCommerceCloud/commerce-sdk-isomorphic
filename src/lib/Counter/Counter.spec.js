import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import Counter from './Counter';

describe('lib > Counter', () => {
  /**
   * Jest hook which runs before each test,
   * @see https://jestjs.io/docs/en/api#beforeeachfn-timeout
   */
  beforeEach(() => {});

  it('renders without crashing', () => {
    /**
     * `asFragment`:
     * @see https://testing-library.com/docs/react-testing-library/api#asfragment
     * `baseElement`:
     * @see https://testing-library.com/docs/react-testing-library/api#baseelement
     */
    const {asFragment, baseElement} = render(<Counter initialValue={6} />);

    /**
     * Basic snapshot test to make sure, that rendered component
     * matches expected footprint.
     */
    expect(asFragment()).toMatchSnapshot();

    /** More precise test for counter value */
    expect(baseElement.querySelector('strong').textContent).toBe('6'); // 6 is value we expect, we need to convert Number to String, because HTMLElement textContent method returns string value
  });

  it('dispatches an action on button click', () => {
    const value = 1;

    /**
     * `getByRole`:
     * @see https://testing-library.com/docs/dom-testing-library/api-queries#byrole
     */
    const {getByRole, baseElement} = render(<Counter initialValue={value} />);

    /**
     * Search for the button and make testing library click on it
     * @see https://testing-library.com/docs/react-testing-library/cheatsheet#events
     */
    fireEvent.click(getByRole('button'));

    /** Check if counter was incremented  */
    expect(baseElement.querySelector('strong').textContent).toBe(
      `${value + 1}`
    );
  });
});
