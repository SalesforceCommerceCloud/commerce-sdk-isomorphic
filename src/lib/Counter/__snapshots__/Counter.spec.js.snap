// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib > Counter renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="counter"
  >
    <h2
      class="header"
    >
      Counter
    </h2>
    <button
      class="button"
      type="button"
    >
      Increment by one
    </button>
    <div>
      Total value: 
      <strong>
        6
      </strong>
    </div>
  </div>
</DocumentFragment>
`;
